### PR TITLE
add decode

### PR DESCRIPTION
**Pull request by liuxin9608**

### CODE REVIEW

File: mmpose/models/keypoint_heads/fc_head.py [hunk 1]
```diff
@@ -115,6 +115,25 @@ def inference_model(self, x, flip_pairs=None):
...
+        Args:
+            img_metas (list(dict)): Information about data augmentation
+                By default this includes:
+                - "image_file: path to the image file
+                - "center": center of the bbox
+                - "scale": scale of the bbox
+                - "rotation": rotation of the bbox
+                - "bbox_score": score of bbox
+            output (np.ndarray[N, K, H, W]): model predicted heatmaps.
+            kwargs: dict contains 'img_size'.
```
Comment (innerlee) [inline mmpose/models/keypoint_heads/fc_head.py:130]: Why not just add `img_size` to argument list?

File: mmpose/models/keypoint_heads/fc_head.py [hunk 2]
```diff
@@ -115,6 +115,25 @@ def inference_model(self, x, flip_pairs=None):
             output_regression = output.detach().cpu().numpy()
         return output_regression
 
+    def decode(self, img_metas, output, **kwargs):
```
Comment (innerlee) [inline mmpose/models/keypoint_heads/fc_head.py:118]: decode_heatmap?

Comment (jin-s13) [reply mmpose/models/keypoint_heads/fc_head.py:118]: We will support multi-task network with multiple heads for depth estimation, classification, and keypoint estimation soon. To make a unified interface for different heads, we will add a `decode` function for all types of heads. 

For keypoint head, `decode` means `decode_heatmap`; while for depth head, `decode` means `decode_depthmap` etc.

Comment (innerlee) [reply mmpose/models/keypoint_heads/fc_head.py:118]: How to determine which `decode_xx` to call?

File: mmpose/models/keypoint_heads/fc_head.py (hunk 1 above)
Comment (innerlee) [reply mmpose/models/keypoint_heads/fc_head.py:130]: Okay this is part of interface refactoring

File: mmpose/models/keypoint_heads/fc_head.py (hunk 2 above)
Comment (innerlee) [reply mmpose/models/keypoint_heads/fc_head.py:118]: I'm trying to understand the design philosophy here. So if you could provide more details...

Comment (innerlee) [reply mmpose/models/keypoint_heads/fc_head.py:118]: Okay, as discussed offline, we rename all decode_xx into decode. There will be no multiple decodes in one class.

File: mmpose/models/keypoint_heads/fc_head.py (hunk 1 above)
Comment (innerlee) [reply mmpose/models/keypoint_heads/fc_head.py:130]: Still not sure about this. Leave it to future revision

**Review by innerlee — APPROVED** [review]

**Pull request merged**